docs/template.md

```diff
@@ -3,4 +3,26 @@ title: "Template"
 weight: 40
 ---
 
->This document is under construction.
\ No newline at end of file
+# Introduction
+
+A Template is a project that specifies what exactly you get as an output of the generation process using AsyncAPI Generator and the [AsyncAPI file](asyncapi-file.md). It is just a set of files where you describe what should be the result of the generation depending on the contents of the AsyncAPI file. 
+
+Some examples of these outputs can be things such as:
+
+- Code
+- Documentation
+- Markdown diagrams
+- Python/Java applications
+
+A template is an independent NodeJS project that’s not related to the `generator` repository. AsyncAPI templates are managed, released, and published separately. You can also create your own templates and manage them on your own.
+
+The generator uses the official NPM library for installing NodeJS dependencies called [Arborist](https://www.npmjs.com/package/@npmcli/arborist). This means templates do not have to be published to package managers to use them, just make sure they have `package.json`.
```

Comment:
```suggestion
The generator uses the official NPM library for installing NodeJS dependencies called [Arborist](https://www.npmjs.com/package/@npmcli/arborist). This means templates do not have to be published to package managers to use them.
```

Reply:
I suggest this as this document should be focused on end user. So end user do not have to make sure template has `package.json` unless they start to write one (for that there is the other document)

so we only need to make sure they are aware they can install templates from different locations